### PR TITLE
Add state dict size and multi model

### PR DESCRIPTION
**Pull request by Hank0626**
1. 新增了统计state_dict中的tesnor和string大小(`MB`)的功能
2. 支持16个resolution的输入图
3. 修复了load不同模型导致的cross_attention_dim维度不匹配的报错问题

### CODE REVIEW

File: examples/unet_torch_interplay.py
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:1]: 这个脚本的本意是提供一个容易阅读的例子。额外的功能是不是不应该直接放进来？

Comment (Hank0626) [reply examples/unet_torch_interplay.py:1]: 那我保留原本的文件，然后这些新加的功能另起一个文件可以吗？

File: examples/unet_torch_interplay.py [hunk 1]
```diff
@@ -106,22 +162,26 @@ def get_arg_meta_of_sizes(batch_sizes, resolution_scales, num_channels):
 @click.option("--repeat", default=100)
 @click.option("--sync_interval", default=50)
 @click.option("--save", is_flag=True)
+@click.option("--with_eager", is_flag=True)
```
Comment (strint) [inline examples/unet_torch_interplay.py:165]: with_eager 改成 without_eager 吧，我看意思是反的

**Comment by Hank0626** [comment]
> 嗯可以import原本的文件里面的函数和类

新加了一个文件`unet_torch_interplay_profile.py`，稍微修改了下原本文件的一些函数接口，使其能被`unet_torch_interplay_profile.py`直接引用

**Review by strint — APPROVED** [review]

File: examples/unet_torch_interplay.py [hunk 2]
```diff
@@ -76,24 +76,25 @@ def save_graph(self, file_path):
...
+    return start + path * i
 
 
 def noise_shape(batch_size, num_channels, image_w, image_h):
     sizes = (image_w // 8, image_h // 8)
     return (batch_size, num_channels) + sizes
 
 
-def get_arg_meta_of_sizes(batch_sizes, resolution_scales, num_channels):
+def get_arg_meta_of_sizes(batch_sizes, resolution_scales, num_channels, cross_attention_dim, start, path):
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:88]: 参数很多的话改为kw参数吧，看的清楚些

File: examples/unet_torch_interplay.py [hunk 3]
```diff
@@ -76,24 +76,25 @@ def save_graph(self, file_path):
         flow.save(state_dict, file_path)
 
 
-def image_dim(i):
-    return 768 + 128 * i
+def image_dim(i, start, path):
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:79]: path是什么意思

File: examples/unet_torch_interplay.py [hunk 4]
```diff
@@ -114,15 +115,17 @@ def benchmark(token, repeat, sync_interval, save, load, file):
     # TODO: reproduce bug caused by changing batch
     # BATCH_SIZES = [4, 2]
 
-    # create a mocked unet graph
     num_channels = 4
-
-    warmup_meta_of_sizes = get_arg_meta_of_sizes(BATCH_SIZES, RESOLUTION_SCALES, num_channels)
-    for (i, m) in enumerate(warmup_meta_of_sizes):
-        print(f"warmup case #{i + 1}:", m)
+    model_id = "runwayml/stable-diffusion-v1-5"
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:119]: 这个如果要增加个变量，就改成一个命令行参数

File: examples/unet_torch_interplay.py (hunk 2 above)
Comment (Hank0626) [reply examples/unet_torch_interplay.py:88]: ok

File: examples/unet_torch_interplay_profile.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    tensor_size, string_size = 0, 0
+    for arg in args_tree.iter_nodes():
+        if isinstance(arg, flow.Tensor):
+            tensor_size += _get_tensor_mem(arg)
+        elif isinstance(arg, str):
+            string_size += len(arg.encode())
+            continue
+    return tensor_size / 1024 / 1024, string_size / 1024 / 1024
+
+class UNetGraphWithCache(flow.nn.Graph):
```
Comment (jackalcooper) [inline examples/unet_torch_interplay_profile.py:51]: 这个代码和上面的版本重复的太多了

File: examples/unet_torch_interplay.py (hunk 3 above)
Comment (Hank0626) [reply examples/unet_torch_interplay.py:79]: path指的是每次增加的步长，要换个什么词吗？

File: examples/unet_torch_interplay.py (hunk 4 above)
Comment (Hank0626) [reply examples/unet_torch_interplay.py:119]: ok

File: examples/unet_torch_interplay_profile.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+@click.command()
+@click.option("--token")
+@click.option("--repeat", default=100)
+@click.option("--sync_interval", default=50)
+@click.option("--save", is_flag=True)
+@click.option("--with_eager", is_flag=True)
+@click.option("--load", is_flag=True)
+@click.option("--file", type=str, default="./unet_graphs")
+@click.option("--model_id", type=str, default="stabilityai/stable-diffusion-2")
+def benchmark(token, repeat, sync_interval, save, with_eager, load, file, model_id):
```
Comment (strint) [inline examples/unet_torch_interplay_profile.py:105]: 这个函数是不是可以可以复用很多之前的代码

File: examples/unet_torch_interplay.py [hunk 7]
```diff
@@ -136,16 +147,14 @@ def benchmark(token, repeat, sync_interval, save, load, file):
 
     time_step = torch.tensor([10]).to("cuda")
     encoder_hidden_states_of_sizes = {
-        batch_size: floats_tensor((batch_size, 77, 768)).to("cuda").to(torch.float16)
+        batch_size: floats_tensor((2, 77, cross_attention_dim)).to("cuda").to(torch.float16)
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:150]: 这里为什么写死了

File: examples/unet_torch_interplay.py [hunk 8]
```diff
@@ -64,37 +71,39 @@ def build(self, latent_model_input, t, text_embeddings):
...
+    
+def noise_shape(batch_size, i, j, num_channels, start, stride):
+    image_w = start + stride * i
+    image_h = start + stride * j
     sizes = (image_w // 8, image_h // 8)
     return (batch_size, num_channels) + sizes
 
 
-def get_arg_meta_of_sizes(batch_sizes, resolution_scales, num_channels):
+def get_arg_meta_of_sizes(batch_sizes, resolution_scales, dim, **kwargs):
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:102]: 这里说的意思是改成 batch_sizes=None, resolution_scales= None, dim= None... 这样完整的列表

File: examples/unet_torch_interplay.py [hunk 9]
```diff
@@ -64,37 +71,39 @@ def build(self, latent_model_input, t, text_embeddings):
...
-    return 768 + 128 * i
-
-
-def noise_shape(batch_size, num_channels, image_w, image_h):
+        state_dict = self.handle_cnt(flow.load(file_path))
+        self.handle_cnt(self.load_runtime_state_dict(state_dict))
+    
+    def save_graph(self, file_path, with_eager=False):
+        state_dict = self.handle_cnt(self.runtime_state_dict(with_eager=with_eager))
+        self.handle_cnt(flow.save(state_dict, file_path))
```
Comment (jackalcooper) [inline examples/unet_torch_interplay.py:87]:
```suggestion
        self.handle_cnt(lambda: flow.save(state_dict, file_path))
```
要这样改吧不然不是个函数？
